proms-resources/backend/src/main/java/io/uhndata/cards/proms/internal/importer/PatientLocalStorage.java

```diff
@@ -0,0 +1,332 @@
+
+            // Update information about the visit ("appointment" is used interchangably here)
+            JsonObject appointmentDetails = this.patientDetails.getJsonObject("nextAppointment");
+            Resource visit = getOrCreateSubject(appointmentDetails.getString("fhirID"),
+                "/SubjectTypes/Patient/Visit/", patient, this.resolver, nodesToCheckin);
+            Resource visitInfo = getOrCreateForm(visit, "/Questionnaires/Visit information/", this.resolver,
+                nodesToCheckin);
+            updateVisitInformationForm(visitInfo, appointmentDetails, this.resolver);
+
+            final Session session = this.resolver.adaptTo(Session.class);
+            session.save();
+
+            final VersionManager vm = session.getWorkspace().getVersionManager();
+            nodesToCheckin.forEach(node -> {
+                try {
+                    vm.checkin(node);
+                } catch (RepositoryException e) {
+                    LOGGER.warn("Failed to check in node {}: {}", node, e.getMessage(), e);
+                }
+            });
+        } catch (IOException e) {
+            LOGGER.error("Could not save patient {}: {}", mrn, e.getMessage(), e);
+        } catch (RepositoryException e) {
+            LOGGER.error("Could not save patient {}: {}", mrn, e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Grab a subject of the specified type, or create it if it doesn't exist.
+     * @param identifier Identifier to use for the subject
+     * @param parentTypePath path to a SubjectType node for this subject
+     * @param parent parent Resource if this is a child of that resource, or null
+     * @param resolver ResourceResolver to use when searching for/creating the node
+     * @param nodesToCheckin Set of nodes that should be checked in after all edits are complete
+     * @return A Subject resource
+     */
+    Resource getOrCreateSubject(String identifier, String parentTypePath, Resource parent, ResourceResolver resolver,
+        Set<String> nodesToCheckin)
+        throws RepositoryException, PersistenceException
+    {
+        Iterator<Resource> patientResource = resolver.findResources(String.format(
+            "SELECT * FROM [cards:Subject] WHERE identifier = \"%s\"", identifier), PatientLocalStorage.JCR_SQL);
+        if (patientResource.hasNext()) {
+            return patientResource.next();
+        } else {
+            Resource parentResource = parent;
+            if (parentResource == null) {
+                parentResource = resolver.getResource("/Subjects/");
+            }
+            Resource patientType = resolver.getResource(parentTypePath);
+            Resource newSubject = resolver.create(parentResource, UUID.randomUUID().toString(), Map.of(
+                PatientLocalStorage.PRIMARY_TYPE, "cards:Subject",
+                "identifier", identifier,
+                "fullidentifier", identifier,
```

Comment:
```suggestion
                "fullIdentifier", identifier,
```

But isn't this going to be computed automatically when saving anyway?